### PR TITLE
Add binary insertion sort algorithm

### PR DESCRIPTION
**Pull request by ryanneilparker**

### CODE REVIEW

File: sorting/binary_insertion_sort.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      return mid
+    elsif val > arr[mid] # val is on the right side
+      start = mid + 1
+    else
+      stop = mid - 1 # val is on the left side
+    end
+  end
+
+  start
+end
```
[the remark this answers is not in the document]
Comment (ryanneilparker) [reply sorting/binary_insertion_sort.rb:18]: Thanks

**Review by ryanneilparker — COMMENTED** [review]
Is it all good to be merged now ?

**Review by vzvu3k6k — APPROVED** [review]
Looks good!

**Comment by vzvu3k6k** [comment]
@ryanneilparker Sorry for the late review. Thank you for your contribution!

**Pull request merged**